src/engine/QueryExecutionTree.cpp

```diff
@@ -139,6 +139,162 @@ nlohmann::json QueryExecutionTree::writeResultAsJson(
   return writeJsonTable(data, offset, limit, validIndices);
 }
 
+// _____________________________________________________________________________
+nlohmann::json QueryExecutionTree::writeResultAsSparqlJson(
+    const vector<string>& selectVars, size_t limit, size_t offset) const {
+  using nlohmann::json;
+
+  // They may trigger computation (but does not have to).
+  shared_ptr<const ResultTable> res = getResult();
+  LOG(DEBUG) << "Resolving strings for finished binary result...\n";
+  vector<pair<size_t, ResultTable::ResultType>> validIndices;
```

Comment:
I find the name `validIndices` confusing given that it's pairs and not just indices.